File: ranla-php/tests/ClientTest.php
<?php

declare(strict_types=1);

namespace Ranla\Tests;

use PHPUnit\Framework\TestCase;
use Ranla\Client;

final class ClientTest extends TestCase
{
    public function test_default_host_is_ranla(): void
    {
        $lastUrl = null;
        $client = new Client(
            'rnl_test_key',
            transport: function (string $method, string $path, ?array $body, array $headers) use (&$lastUrl): array {
                $lastUrl = $path;
                return ['id' => 'msg_1', 'status' => 'sent'];
            },
        );

        $result = $client->emails->send([
            'from' => 'a@example.com',
            'to' => 'b@example.com',
            'subject' => 'Hi',
            'html' => '<p>Hi</p>',
        ]);

        $this->assertSame(['id' => 'msg_1', 'status' => 'sent'], $result);
        $this->assertSame('/emails', $lastUrl);
    }

    public function test_base_url_override(): void
    {
        $client = new Client(
            'rnl_test_key',
            'https://api.example.com',
            transport: static fn (): array => ['ok' => true],
        );

        $this->assertSame(['ok' => true], $client->request('GET', '/health'));
    }
}

File: go/ranla/client_test.go
package ranla_test

import (
	"encoding/json"
	"net/http"
	"net/http/httptest"
	"testing"

	ranla "github.com/Super-Send/supersendtx-sdks/go/ranla"
)

func TestNewDefaultsToRanlaHost(t *testing.T) {
	server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
		if r.URL.Path != "/emails" || r.Method != http.MethodPost {
			t.Fatalf("unexpected request: %s %s", r.Method, r.URL.Path)
		}
		if got := r.Header.Get("Authorization"); got != "Bearer rnl_test" {
			t.Fatalf("unexpected auth header: %q", got)
		}
		_ = json.NewEncoder(w).Encode(map[string]any{"id": "msg_1", "status": "sent"})
	}))
	defer server.Close()

	client, err := ranla.NewClient("rnl_test", server.URL, server.Client())
	if err != nil {
		t.Fatal(err)
	}

	result, err := client.Emails.Send(map[string]any{
		"from":    "a@example.com",
		"to":      "b@example.com",
		"subject": "Hi",
		"html":    "<p>Hi</p>",
	})
	if err != nil {
		t.Fatal(err)
	}
	if result["id"] != "msg_1" {
		t.Fatalf("unexpected result: %#v", result)
	}
}

func TestNewUsesRanlaHostByDefault(t *testing.T) {
	if ranla.DefaultAPIBaseURL != "https://api.ranla.ai" {
		t.Fatalf("unexpected default host: %q", ranla.DefaultAPIBaseURL)
	}

	client, err := ranla.New("rnl_test")
	if err != nil {
		t.Fatal(err)
	}
	if client.BaseURL != ranla.DefaultAPIBaseURL {
		t.Fatalf("unexpected base url: %q", client.BaseURL)
	}
}

func TestNewRejectsInvalidPrefix(t *testing.T) {
	if _, err := ranla.New("bad"); err == nil {
		t.Fatal("expected error for invalid api key prefix")
	}
}
